### PR TITLE
[TASK] Add TS linting and PHP code sniffing

### LINKED ISSUE #26: Add PSR-12 code sniffing as a GitHub action

### PR DESCRIPTION
**Pull request by andreaswolf**
Resolves: #26, #27

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
     },
     "require": {
         "php": "~7.2 || ~7.3 || ~7.4",
+        "helmich/typo3-typoscript-lint": "^2.0",
+        "squizlabs/php_codesniffer": "^3.5",
```
Comment (oliverklee) [inline composer.json]: For dev dependencies, I prefer to point to the latest bugfix release, i.e., in this case ^2.0.1 and ^3.5.3 so we won't get the added complexity of bugs with older versions.

File: composer.json [hunk 2]
```diff
@@ -48,8 +50,12 @@
     },
     "scripts": {
         "ci:php:lint": "find *.php Classes/ Configuration/ Tests/ -name '*.php' -print0 | xargs -0 -n 1 -P 4 php -l",
+        "ci:php:sniff": "phpcs Classes Tests",
+        "ci:ts:lint": "typoscript-lint -c Configuration/TsLint.yml --ansi -n --fail-on-warnings -vvv Configuration/TypoScript/",
```
Comment (oliverklee) [inline composer.json:54]: We can drop the trailing slash here.

**Review by oliverklee — CHANGES_REQUESTED** [review]

**Review by oliverklee — CHANGES_REQUESTED** [review]
Also, please let's have a changelog line for this.

File: composer.json (hunk 2 above)
Comment (andreaswolf) [reply composer.json:54]: Since it's a folder, I would prefer to keep that trailing slash.

Comment (oliverklee) [reply composer.json:54]: IIRC, some tools automatically add the slash, leading to double slashes in the output in cases of validation errors. (I don't remember which of the tools I use in CI were affected, though.)

Comment (oliverklee) [inline composer.json:54]: The configuration file `Configuration/TsLint.yml` does not exist yet.

File: composer.json [hunk 3]
```diff
@@ -48,8 +50,12 @@
     },
     "scripts": {
         "ci:php:lint": "find *.php Classes/ Configuration/ Tests/ -name '*.php' -print0 | xargs -0 -n 1 -P 4 php -l",
+        "ci:php:sniff": "phpcs Classes Tests",
```
Comment (oliverklee) [inline composer.json:53]: We also need the configuraton file `phpcs.xml.dist` for this to work correctly.

**Review by oliverklee — CHANGES_REQUESTED** [review]

**Review by oliverklee — APPROVED** [review]

**Review by oliverklee — APPROVED** [review]

**Pull request merged**